test/playlist-selectors.test.js

```diff
@@ -0,0 +1,52 @@
+import { module, test } from 'qunit';
+import { movingAverageBandwidthSelector } from '../src/playlist-selectors';
+import Config from '../src/config';
+
+const video = document.createElement('video');
+const hls = {
+  tech_: {
+    el() {
+      return video;
+    }
+  },
+  playlists: {
+    master: {
+      playlists: []
+    }
+  }
+};
+
+module('Playlist Selectors');
+
+test('Exponential moving average has a configurable decay parameter', function(assert) {
+  let playlist;
+  const instantAverage = movingAverageBandwidthSelector(1.0);
+
+  hls.playlists.master.playlists = [
+    { attributes: { BANDWIDTH: 1 } },
+    { attributes: { BANDWIDTH: 50 } },
+    { attributes: { BANDWIDTH: 100 } }
+  ];
+  hls.systemBandwidth = 50 * Config.BANDWIDTH_VARIANCE + 1;
+  playlist = instantAverage.bind(hls)();
+  assert.equal(playlist.attributes.BANDWIDTH, 50, 'selected the middle playlist');
+
+  hls.systemBandwidth = 100 * Config.BANDWIDTH_VARIANCE + 1;
+  playlist = instantAverage.bind(hls)();
+  assert.equal(playlist.attributes.BANDWIDTH, 100, 'selected the top playlist');
+
+  const fiftyPercentDecay = movingAverageBandwidthSelector(0.5);
+
+  hls.systemBandwidth = 100 * Config.BANDWIDTH_VARIANCE + 1;
+  playlist = fiftyPercentDecay.bind(hls)();
+  assert.equal(playlist.attributes.BANDWIDTH, 100, 'selected the top playlist');
+
+  // bandwidth = 0.5 * systemBandwidth + 0.5 * (100 * variance + 1)
```

Comment:
Minor, but might be worth adding ` average = decay * systemBandwidth + (1 - decay) * average` above